pennylane/ops/qubit/observables.py

```diff
@@ -152,12 +166,33 @@ def num_params(self):
     def label(self, decimals=None, base_label=None):
         return super().label(decimals=decimals, base_label=base_label or "𝓗")
 
-    @classmethod
-    def _matrix(cls, *params):
-        A = params[0]
-        if not isinstance(A, coo_matrix):
+    @staticmethod
+    def compute_matrix(H):
+        """Canonical matrix representation of this operator.
+
+        Args:
+            H (scipy.sparse.coo.coo_matrix): sparse matrix used to create this operator
+
+        Returns:
+            scipy.sparse.coo.coo_matrix: matrix representation
+
+        **Example**
+
+        >>> from scipy.sparse import coo_matrix
+        >>> H = np.array([[6+0j, 1-2j],[1+2j, -1]])
+        >>> H = coo_matrix(H)
+        >>> res = qml.SparseHamiltonian.compute_matrix(H)
+        (0, 0)	(6+0j)
+        (0, 1)	(1-2j)
+        (1, 0)	(1+2j)
+        (1, 1)	(-1+0j)
+        >>> type(res)
+        <class 'scipy.sparse.coo.coo_matrix'>
+        """
+        # ToDo[Maria/Josh]: return H.toarray() and add a separate `sparse_matrix` method
+        if not isinstance(H, coo_matrix):
             raise TypeError("Observable must be a scipy sparse coo_matrix.")
-        return A
+        return H
```

Comment:
For now I keep this consistent with the status quo. In a follow-up PR, this method will convert the sparse to a dense matrix, and `sparse_matrix` will return the original one.